direct/core/src/main/java/cz/o2/proxima/direct/batch/BatchLogReaders.java

```diff
@@ -40,8 +42,11 @@
    * @return throughput limited {@link BatchLogReader}
    */
   public static BatchLogReader withLimitedThroughput(
-      BatchLogReader delegate, ThroughputLimiter limiter) {
-    return new ThroughputLimitedBatchLogReader(delegate, limiter);
+      BatchLogReader delegate, @Nullable ThroughputLimiter limiter) {
```

Comment:
I'd rather see Objects.requireNonNull here... It basically doesn't make any sense to call this method will null limiter

Reply:
makes, it is called with null limiter when there is no limiter configured

Reply:
Then it shouldn't be called at all. The null check should be placed "one level higher" to make this more error prone.

Reply:
Strongly disagree. There is nothing neither confusing nor error prone in using `null` values in APIs, if used semantically correct. This method is supposed to return a limited log reader, which - when seeded with null limiter - is semantically correct to the original reader. This invariant is what it makes clear and absolutely well defined to use the null value exactly this way.